scrabble.js

```diff
@@ -1,8 +1,204 @@
-var Scrabble = function() {};
 
-// YOUR CODE HERE
-Scrabble.prototype.helloWorld = function() {
-  return 'hello world!';
-};
+  var prompt = require('prompt');
+  var request = require('request');
 
+// SCRABBLE
+  Scrabble = function() {
+    this._points = {
+      A: 1,
+      B: 3,
+      C: 3,
+      D: 2,
+      E: 1,
+      F: 4,
+      G: 2,
+      H: 4,
+      I: 1,
+      J: 8,
+      K: 5,
+      L: 1,
+      M: 3,
+      N: 1,
+      O: 1,
+      P: 3,
+      Q: 10,
+      R: 1,
+      S: 1,
+      T: 1,
+      U: 1,
+      V: 4,
+      W: 4,
+      X: 8,
+      Y: 4,
+      Z: 10
+    };
+  };
+  Scrabble.prototype.score = function(word){
+    var score = 0;
+    var tiles = 0;
+
+    var wordArray = word.toUpperCase().split("");
+
+    for(var char of wordArray){
+      score += this._points[char];
+
+      tiles += 1;
+    };
+
+    if(tiles == 7){
+      score += 50;
+    };
+    return score;
+  };
+
+  Scrabble.prototype.highestScoreFrom = function(arrayOfWords){
+
+    var maxWord = [arrayOfWords[0], this.score(arrayOfWords[0])];
+    var currentScore = this.score(arrayOfWords[0]);
```

Comment:
Right! That's the benefit of maxWord.

On Wed, Jun 15, 2016 at 9:32 PM, Jennie Buechner <notifications@github.com>
wrote:

> In scrabble.js
> <https://github.com/Ada-C5/js-scrabble/pull/17#discussion_r67286124>:
>
> > +    for(var char of wordArray){
> > +      score += this._points[char];
> > +
> > +      tiles += 1;
> > +    };
> > +
> > +    if(tiles == 7){
> > +      score += 50;
> > +    };
> > +    return score;
> > +  };
> > +
> > +  Scrabble.prototype.highestScoreFrom = function(arrayOfWords){
> > +
> > +    var maxWord = [arrayOfWords[0], this.score(arrayOfWords[0])];
> > +    var currentScore = this.score(arrayOfWords[0]);
>
> Nitpick - It's not efficient to call the score function twice on the same
> word.
>
> Could just do currentScore = maxWord[1] for example
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Ada-C5/js-scrabble/pull/17/files/68f6a6ae8a0ccd97a9fe4f3a205163ad6eefb2ee#r67286124>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AOaZbfd5y9-7Or8snk-oqvTUn3kRFZ_Xks5qMNHsgaJpZM4Iz-uw>
> .
>